src/rules/no-unnecessary-components.js

```diff
@@ -68,33 +70,36 @@ const rule = ESLintUtils.RuleCreator.withoutDocs({
         const isPrimer = skipImportCheck || isPrimerComponent(name, context.sourceCode.getScope(openingElement))
         if (!isPrimer) return
 
-        // Validate the attributes and ensure an `sx` prop is present or spreaded in
+        /** @param {string} name */
+        const isAllowedProp = name => componentConfig.allowedProps.has(name) || isStyledSystemProp(name)
+
+        // Validate the attributes and ensure an allowed prop is present or spreaded in
         /** @type {typeof attributes[number] | undefined | null} */
         let asProp = undefined
         for (const attribute of attributes) {
-          // If there is a spread type, check if the type of the spreaded value has an `sx` property
+          // If there is a spread type, check if the type of the spreaded value has an allowed property
           if (attribute.type === 'JSXSpreadAttribute') {
             const services = ESLintUtils.getParserServices(context)
             const typeChecker = services.program.getTypeChecker()
 
             const spreadType = services.getTypeAtLocation(attribute.argument)
-            if (typeChecker.getPropertyOfType(spreadType, 'sx') !== undefined) return
 
-            // Check if the spread type has a string index signature - this could hide an `sx` property
+            // Check if the spread type has a string index signature - this could hide an allowed property
             if (typeChecker.getIndexTypeOfType(spreadType, IndexKind.String) !== undefined) return
```

Comment:
I switched these checks around since theoretically `getIndexOfType` should be slightly faster than `getPropertiesOfType` so we can potentially short-circuit a little bit faster and save some time.